pandas/core/arrays/period.py

```diff
@@ -698,15 +705,17 @@ def asfreq(self, freq=None, how: str = "E") -> Self:
         PeriodIndex(['2010', '2011', '2012', '2013', '2014', '2015'],
         dtype='period[A-DEC]')
 
-        >>> pidx.asfreq('M')
+        >>> pidx.asfreq('ME')
         PeriodIndex(['2010-12', '2011-12', '2012-12', '2013-12', '2014-12',
         '2015-12'], dtype='period[M]')
 
-        >>> pidx.asfreq('M', how='S')
+        >>> pidx.asfreq('ME', how='S')
         PeriodIndex(['2010-01', '2011-01', '2012-01', '2013-01', '2014-01',
         '2015-01'], dtype='period[M]')
         """
         how = libperiod.validate_end_alias(how)
+        if not isinstance(freq, int):
+            freq = OFFSET_TO_PERIOD_FREQSTR.get(freq, freq)
```

Comment:
also, it's possible to do `asfreq('2M')`, so we should probably use the new function here (rather than the dict)